src/Stratis.Features.FederatedPeg/Wallet/FederationWalletManager.cs

```diff
@@ -859,6 +859,16 @@ public bool ValidateTransaction(Transaction transaction, bool checkSignature = f
                 if (!this.TransactionHasValidUTXOs(transaction, coins))
                     return false;
 
+                // Verify that there are no earlier unspent UTXOs.
```

Comment:
I've reintroduced this code in this PR but I can't see how the nodes can get into a situation where this wouldn't be true... As we are ordering by time and only using confirmations=1, it's always going to contain the "first" combination of UnspentOutputs that adds to the total.

Reply:
My guess is this used to serve a purpose when the sidechains team didn't anticipate that the CoinSelector was reordering their coins?

Reply:
If a federation peer mines a block it would temporarily have an additional UTXO that other peers would not have. As a result (if confirmations < 1) it produces the next withdrawal transaction with that UTXO included while the other peers do not do likewise. Those other peers need a means of recovery which I think is what this code would achieve. 

I.e. it caters for peers not being equally up to date when transactions are generated.

Of course confirmation > 0 so we would not expect this situation to arise but maybe it can if reorgs are involved.